src/allmydata/test/test_system.py

```diff
@@ -657,7 +658,23 @@ def _check_download_5(res):
             self.failUnlessEqual(res, NEWERDATA)
         d.addCallback(_check_download_5)
 
-        def _corrupt_shares(res):
+        @async_to_deferred
+        async def _check_write_at_offset(newnode):
```

Comment:
A comment or docstring about the intent of this new set of actions would be nice.